sdk/core/azure-core/src/datetime.cpp

```diff
@@ -729,42 +729,39 @@ DateTime DateTime::Parse(std::string const& dateTime, DateFormat format)
   }
 }
 
-std::string DateTime::ToString(DateFormat format, TimeFractionFormat fractionFormat) const
+void DateTime::ValidateYear() const
 {
-  {
-    static DateTime const Year0001 = DateTime();
-    static DateTime const Eoy9999 = DateTime(9999, 12, 31, 23, 59, 59, 9999999, -1, 0, 0);
-
-    auto outOfRange = 0;
-    if (*this < Year0001)
-    {
-      outOfRange = -1;
-    }
-    else if (*this > Eoy9999)
-    {
-      outOfRange = +1;
-    }
+  static DateTime const Year0001 = DateTime();
+  static DateTime const Eoy9999 = DateTime(9999, 12, 31, 23, 59, 59, 9999999, -1, 0, 0);
```

Comment:
Do we want this as constexpr? in the unnamed namespace?

Reply:
I don't want to modify implementation details in this PR.